### PR TITLE
test: small change in common method

### PR DESCRIPTION
**Pull request by NandanAnantharamu**
/ok-to-test tags="@tag.All"

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9764002269>
> Commit: 9d629c9dd28abe15aadd859dc68c079462539bc8
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9764002269&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved reliability of waiting for elements to appear on the Home Page by adding a conditional check, reducing potential errors related to undefined elements.

### CODE REVIEW

File: app/client/cypress/support/Pages/HomePage.ts [hunk 1]
```diff
@@ -522,7 +522,7 @@ export class HomePage {
       "not.include",
       "edit",
     );
-    this.agHelper.WaitUntilEleAppear(element ?? this.locator._backToEditor);
+    if (element) this.agHelper.WaitUntilEleAppear(this.locator._backToEditor);
```
Comment (ApekshaBhosale) [inline app/client/cypress/support/Pages/HomePage.ts:525]: @NandanAnantharamu is the condition right? if element is present then wait for till elem appear?

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/cypress/support/Pages/HomePage.ts (hunk 1 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/HomePage.ts:525]: @ApekshaBhosale earlier condition was working even without element, added this to make sure the page is loaded completely

Comment (ApekshaBhosale) [reply app/client/cypress/support/Pages/HomePage.ts:525]: but isn't condition should be if element is not present then wait for it to appear? what is the point of wait untill appear if element is already there @NandanAnantharamu

Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/HomePage.ts:525]: ok let me update with AssertElement visibility and then click on backToEditor

Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/HomePage.ts:525]: @ApekshaBhosale so keeping the changes as is as discussed as its a necessity for one of the tests we have

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**